rbi/stdlib/rubygems.rbi

```diff
@@ -2634,27 +2634,36 @@ class Gem::Version < Object
 
   sig do
     params(
-      version: T.any(String, Gem::Version),
-    )
-    .void
+      version: T.any(
+        String,
+        Integer,
+        Float, # Technically, any `to_s`able object would work.
```

Comment:
E.g. `Gem::Version(1)` and `Gem::Version(1.23)` are allowed.

Does Sorbet have a way to express structural requirements about arguments? E.g. anything that responds to `#to_s`?

Reply:
Sorbet does not support something like `T::RespondsTo[:to_s]`, and I suspect never will.

https://sorbet.org/docs/flow-sensitive#what-about-respond_to